src/main/kotlin/calculator/Operator.kt

```diff
@@ -0,0 +1,24 @@
+package calculator
+
+private const val DOUBLE_ZERO = 0.0
+
+enum class Operator(private val value: String, private val function: (Double, Double) -> Double) {
+    PLUS("+", { t, u -> t + u }),
+    MINUS("-", { t, u -> t - u }),
+    DIVISION("/", { t, u ->
+        require(u != DOUBLE_ZERO) { "0으로 나눌수 없습니다." }
+        t / u
+    }),
+    TIMES("*", { t, u -> t * u });
+
+    companion object {
+        fun find(operator: String): Operator {
+            return values().firstOrNull { it.value == operator }
+                ?: throw IllegalArgumentException("사칙연산 기호가 올바르지 않습니다.")
+        }
+    }
+
+    fun operate(first: Double, second: Double): Double {
+        return function.invoke(first, second)
```

Comment:
```suggestion
    fun operate(first: Double, second: Double): Double {
        return function(first, second)
```

코틀린에는 이 것도 연산자 오버로딩인가 싶은 것들이 몇 가지 있습니다.
대표적인 것이 invoke -> (), get -> [] 등이 있을 것 같아요.

Reply:
invoke 도 제거할수 있네요 😄